===================================================================
RCS file: /home/cvspsrv/cvsroot/sound/ecasound/configure.in,v
retrieving revision 1.152
retrieving revision 1.158
diff -u -r1.152 -r1.158
--- configure.in	2003/04/02 20:14:10	1.152
+++ configure.in	2003/05/30 19:28:47	1.158
@@ -1,6 +1,6 @@
 dnl ---
 dnl configure.in for ecasound
-dnl last modified: 20030402-14
+dnl last modified: 20030530-20
 dnl 
 dnl Many of the tests and macros in this file have been borrowed from 
 dnl other GPL-lisenced packages (Alsaplayer, ALSA, SIP, PyQt, 
@@ -23,7 +23,7 @@
 dnl ---
 dnl Init automake
 dnl ---
-AM_INIT_AUTOMAKE(ecasound, 2.2.3)
+AM_INIT_AUTOMAKE(ecasound, 2.2.4)
 AM_CONFIG_HEADER(config.h)
 
 dnl ---
@@ -32,15 +32,17 @@
 dnl defines: LIBECASOUND_VERSION, LIBECASOUNDC_VERSION, LIBKVUTILS_VERSION
 dnl ---
 
-LIBECASOUND_VERSION=10
-LIBECASOUND_VERSION_AGE=1
+LIBECASOUND_VERSION=11
+LIBECASOUND_VERSION_AGE=2
 LIBECASOUNDC_VERSION=1
-LIBKVUTILS_VERSION=5
-LIBKVUTILS_VERSION_AGE=1
+LIBECASOUNDC_VERSION_AGE=0
+LIBKVUTILS_VERSION=6
+LIBKVUTILS_VERSION_AGE=2
 
 AC_SUBST(LIBECASOUND_VERSION)
 AC_SUBST(LIBECASOUND_VERSION_AGE)
 AC_SUBST(LIBECASOUNDC_VERSION)
+AC_SUBST(LIBECASOUNDC_VERSION_AGE)
 AC_SUBST(LIBKVUTILS_VERSION)
 AC_SUBST(LIBKVUTILS_VERSION_AGE)
 
@@ -432,39 +434,46 @@
 dnl Checks for pyecasound
 dnl
 dnl defines: ECA_S_PYTHON_INCLUDES, ECA_S_PYTHON_DLMODULES, 
-dnl          ECA_AM_COMPILE_PYECASOUND, ECA_AM_HAVE_PYTHON
+dnl          ECA_AM_PYECASOUND_CEXT, ECA_AM_PYECASOUND_INSTALL
 dnl --
 
-compile_pyecasound=no
+pyecasound_support=python
 
 AC_ARG_ENABLE(pyecasound,
-	      [  --enable-pyecasound 	  Enable compilation of pyecasound (default = no)],
+	      [  --enable-pyecasound 	  Enable compilation of pyecasound (default = python)],
 	      [
 	      case "$enableval" in
 	        y | yes)
-		  AC_MSG_RESULT(yes)
-		  compile_pyecasound=yes
+		  pyecasound_support=c
 		;;
 
-		n | no)
-		  AC_MSG_RESULT(no)
-		  compile_pyecasound=no
+		n | no | none)
+		  pyecasound_support=none
+		;;
+
+		python)
+		  pyecasound_support=python
+		;;
+
+		c)
+		  pyecasound_support=c
 		;;
         
 		*)
-		  AC_MSG_ERROR([Invalid parameter value for --enable-pyecasound: $enableval])
+		  pyecasound_support=python
 		;;
 	      esac
 	      ])
 
 dnl --
-dnl Python interpreter
+dnl Python interpreter and installation prefix
 
 AC_PATH_PROG(PYTHONPATH,python,none)
 if test x$PYTHONPATH = xnone; then
-  python_support=no
+  pyecasound_support=none
+  python_install_prefix=""
 else
-  python_support=yes
+  python_install_prefix=`python -c "import sys; print (sys.prefix)"`
 fi
 
 dnl --
@@ -497,7 +506,10 @@
 if test "X$ECA_S_PYTHON_INCLUDES" = "X"
 then
 	AC_MSG_WARN([Python.h not found])
-	compile_pyecasound=no
+	if test x${pyecasound_support} = xc; then
+	    dnl -- No python headers found, cannot compile C extension
+	    pyecasound_support=python
+	fi
 else
 	AC_MSG_RESULT($ECA_S_PYTHON_INCLUDES)
 fi
@@ -511,14 +523,35 @@
 
 if test "X$with_python_modules" = "X"
 then
-    pymoddirs="/usr/local/lib /usr/lib"
-    dnl -- Double-brackets to espace the real brackets
-    pymoddirsmore=`python -c "import sys; print (sys.prefix + '/lib/python' + sys.version[[:3]])"`
-    pymoddirs="$pymoddirs $pymoddirsmore"
+    if test x${python_install_prefix} != x${ecaprefix} -a \
+            x${ecaprefix} != x/usr -a \
+	    x${ecaprefix} != x/usr/local ; then
+        pyecasound_support=none
+	AC_MSG_RESULT([none])
+	if test x$PYTHONPATH != xnone; then
+	    python_prefix_tmp=`python -c "import sys; print (sys.prefix + '/lib/python' + sys.version[[:3]])"`
+	else
+	    python_prefix_tmp="DIR"
+	fi
+        AC_MSG_WARN([
+	***
+	*** Python modules won't be installed as the module install prefix does
+	*** not match ecasound's build prefix! To override, set python
+	*** module directory explicitly (for example 
+	*** "--with-python-modules=${python_prefix_tmp}").
+	***
+	])
+    else
+        pymoddirs="/usr/local/lib /usr/lib"
+        dnl -- Double-brackets to espace the real brackets
+        pymoddirsmore=`python -c "import sys; print (sys.prefix + '/lib/python' + sys.version[[:3]])"`
+        pymoddirs="$pymoddirs $pymoddirsmore"
+	AC_MSG_RESULT($pymoddirs)
+    fi
 else
     pymoddirs=$with_python_modules
+    AC_MSG_RESULT($pymoddirs)
 fi
-AC_MSG_RESULT($pymoddirs)
 
 dnl ---
 dnl a hack to support broken debian python installs
@@ -555,9 +588,9 @@
 AC_SUBST(ECA_S_PYTHON_MODULES)
 AC_SUBST(ECA_S_PYTHON_DLMODULES)
 
-dnl if test x$disable_pyecasound = xyes ; then compile_pyecasound=no ; fi
-AM_CONDITIONAL(ECA_AM_COMPILE_PYECASOUND, test x$compile_pyecasound = xyes)
-AM_CONDITIONAL(ECA_AM_HAVE_PYTHON, test x$python_support = xyes)
+dnl if test x$disable_pyecasound = xyes ; then pyecasound_support=no ; fi
+AM_CONDITIONAL(ECA_AM_PYECASOUND_CEXT, test x$pyecasound_support = xc)
+AM_CONDITIONAL(ECA_AM_PYECASOUND_INSTALL, test x$pyecasound_support != xnone)
 
 dnl ------------------------------------------------------------------
 
@@ -621,12 +654,12 @@
 	      case "$enableval" in
 	        y | yes)
 		  AC_MSG_RESULT(yes)
-		  termcap_library=yes
+		  termcap_library=ncurses
 		;;
 
 		n | no)
 		  AC_MSG_RESULT(no)
-		  termcap_library=no
+		  termcap_library=none
 		;;
         
 		*)
@@ -788,8 +821,14 @@
      [AC_CHECK_HEADER(readline/history.h,,
        AC_MSG_ERROR([*** readline history headers not installed ***]))])
 
-  AC_CHECK_LIB(readline, main,, AC_MSG_ERROR([*** readline support not installed ***]))
-  AC_CHECK_LIB(history, main,, AC_MSG_ERROR([*** readline history support not installed ***]))
+  if test x${termcap_library} = xnone; then
+     readline_extra_libs="";
+  else
+     readline_extra_libs="-l${termcap_library}"
+  fi
+
+  AC_CHECK_LIB(readline, main, readline_library="-lreadline", AC_MSG_ERROR([*** readline support not installed ***]), ${readline_extra_libs})
+  AC_CHECK_LIB(history, main, readline_library="${readline_library} -lhistory", AC_MSG_ERROR([*** readline history support not installed ***]), ${readline_extra_libs})
   AC_SUBST(ECA_S_READLINE_INCLUDES)
 else
   AC_CONFIG_SUBDIRS(readline-4.0)
@@ -865,6 +904,9 @@
 AC_CHECK_FUNCS(sched_getparam)
 AC_CHECK_FUNCS(sched_getscheduler)
 AC_CHECK_FUNCS(sched_setscheduler)
+AC_CHECK_FUNCS(pthread_self)
+AC_CHECK_FUNCS(pthread_getschedparam)
+AC_CHECK_FUNCS(pthread_setschedparam)
 AC_CHECK_FUNCS(sigprocmask)
 AC_CHECK_FUNCS(sigwait)
 
@@ -943,8 +985,12 @@
 echo "Following packages are now configured:"
 echo "ecasound: 		yes (ecasound,libecasound,libecasoundc)"
 echo "ecatools: 		yes"
-if test x$compile_pyecasound = xyes ; then
-	echo "pyecasound: 		yes"
+if test x$pyecasound_support != xnone ; then
+	if test x$pyecasound_support = xc ; then
+		echo "pyecasound: 		yes (as Python C extension module)"
+	else
+		echo "pyecasound: 		yes (as native Python module)"
+	fi
 else
 	echo "pyecasound: 		no"
 fi
@@ -1011,7 +1057,7 @@
 echo "-----------------------------------------------------------------"
 echo "Following directories are used:"
 echo "Directory prefix:    "$ECA_S_PREFIX
-if test x$compile_pyecasound = xyes ; then
+if test x$pyecasound_support != xnone ; then
 	echo "Python include dir:	"$ECA_S_PYTHON_INCLUDES
 	echo "Python module dir:	"$ECA_S_PYTHON_MODULES
 fi
